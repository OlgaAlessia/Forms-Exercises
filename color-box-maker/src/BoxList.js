import React, { useState } from "react";
import NewBoxForm from "./NewBoxForm";
import "./BoxList.css";
import Box from "./Box";
import { v4 as uuid } from "uuid";

function BoxList() {
    const [boxes, setBoxes] = useState([]);


    function handleRemove(id) {
        setBoxes(boxes.filter(box => box.id !== id));
    }


    /** Add new box object to boxList. */
    const addBox = box => {
        let newBox = { ...box, id: uuid() };
        setBoxes(boxes => [...boxes, newBox]);
    };
    // end addBox

    return (
        <div>
            <h3>Box List</h3>
            <NewBoxForm addBox={addBox} />
            <div className="BoxList">
                {boxes.map(({ id, color, width, height }) => <Box color={color} width={width} height={height} key={id} handleRemove={evt => handleRemove(id)} />)}
            </div>
        </div>
    );
};
// end

export default BoxList;
