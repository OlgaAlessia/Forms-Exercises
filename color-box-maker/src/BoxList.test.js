import React from "react";
import { render, fireEvent } from "@testing-library/react";
import BoxList from "./BoxList";

it("renders without crashing", function () {
  render(<BoxList />);
});

it("matches snapshot", function () {
  const { asFragment } = render(<BoxList />);
  expect(asFragment()).toMatchSnapshot();
});

it("should add new box", function () {
  const { queryByText, getByLabelText } = render(<BoxList />);
  const input = getByLabelText("Color:");
  const btn = queryByText("Add a new box!");
  //expect(queryByText('Walk the dog')).not.toBeInTheDocument();
  //expect(queryByText('Walk the dog')).not.toBeInTheDocument();
  //fireEvent.change(input, { target: { value: 'Walk the dog' } });
  //fireEvent.click(btn);
  //expect(queryByText('Walk the dog')).toBeInTheDocument();
})


