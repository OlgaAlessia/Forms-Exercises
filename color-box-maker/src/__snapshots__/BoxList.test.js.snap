// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <h3>
      Box List
    </h3>
    <form
      class="NewBoxForm"
    >
      <label
        for="color"
      >
        Color:
      </label>
      <input
        id="color"
        name="color"
        type="text"
        value=""
      />
      <label
        for="width"
      >
        Width:
      </label>
      <input
        id="width"
        name="width"
        type="number"
        value="0"
      />
      <label
        for="height"
      >
        Height:
      </label>
      <input
        id="height"
        name="height"
        type="number"
        value="0"
      />
      <br />
      <button>
        Add a new box!
      </button>
    </form>
    <div
      class="BoxList"
    />
  </div>
</DocumentFragment>
`;
